src/coord/src/coord.rs

```diff
@@ -258,6 +261,7 @@ pub struct Config {
     pub metrics_registry: MetricsRegistry,
     pub persister: PersisterWithConfig,
     pub now: NowFn,
+    pub secrets_storage_path: PathBuf,
```

Comment:
We're going to want to provide the `Box<dyn SecretsController>` here rather than the path so that the type of secrets controller can be conditional on the `--orchestrator=kubernetes` flag. I recommend doing the hoisting now, and making materialized/lib.rs responsible for unconditionally creating the `FilesystemController`!